src/frontend/staff-portal/src/app/components/jj-dispute-info/jj-count/jj-count.component.ts

```diff
@@ -1,381 +1,445 @@
+          _withdrawn: this.jjDisputedCount.jjDisputedCountRoP.withdrawn === this.Withdrawn.Y,
+          _abatement: this.jjDisputedCount.jjDisputedCountRoP.abatement === this.Abatement.Y
+        });
+      }
+    }
+  }
+
+  private initListeners() {
+    if (this.jjDisputedCount) {
+      // listen for form changes
+      this.form.valueChanges.subscribe(() => {
+        Object.assign(this.jjDisputedCount, this.form.getRawValue()); // get raw value includes disabled fields
+        if (this.jjDisputedCount.latestPleaUpdateTs) {
+          this.jjDisputedCount.latestPleaUpdateTs = new Date(this.jjDisputedCount.latestPleaUpdateTs).toISOString();
+        }
+        this.jjDisputedCount.includesSurcharge = (this.inclSurcharge === "yes" ? this.IncludesSurcharge.Y : this.IncludesSurcharge.N);
+        this.jjDisputedCountUpdate.emit(this.jjDisputedCount);
+      });
+
+      this.countForm.valueChanges.subscribe(() => {
+        this.jjDisputedCount = { ...this.jjDisputedCount, ...this.countForm.value };
+        this.inclSurcharge = (this.jjDisputedCount.includesSurcharge === this.IncludesSurcharge.Y ? "yes" : "no");
+        this.jjDisputedCountUpdate.emit(this.jjDisputedCount);
+      });
+
+      this.countRoPForm.valueChanges.subscribe(() => {
+        this.jjDisputedCount.jjDisputedCountRoP = { ...this.jjDisputedCount.jjDisputedCountRoP, ...this.countRoPForm.value };
+        this.jjDisputedCount.jjDisputedCountRoP.jailIntermittent = this.countRoPForm.get("_jailIntermittent").value ? this.JailIntermittent.Y : this.JailIntermittent.N;
+        this.jjDisputedCount.jjDisputedCountRoP.dismissed = this.countRoPForm.get("_dismissed").value ? this.Dismissed.Y : this.Dismissed.N;
+        this.jjDisputedCount.jjDisputedCountRoP.forWantOfProsecution = this.countRoPForm.get("_forWantOfProsecution").value ? this.ForWantOfProsecution.Y : this.ForWantOfProsecution.N;
+        this.jjDisputedCount.jjDisputedCountRoP.withdrawn = this.countRoPForm.get("_withdrawn").value ? this.Withdrawn.Y : this.Withdrawn.N;
+        this.jjDisputedCount.jjDisputedCountRoP.abatement = this.countRoPForm.get("_abatement").value ? this.Abatement.Y : this.Abatement.N;
+        this.jjDisputedCountUpdate.emit(this.jjDisputedCount);
+      });
+    }
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes?.jjDisputeInfo?.currentValue) {
+      this.jjDisputeInfo = { ...this.jjDisputeInfo, ...this.jjDisputeInfo };
+      this.initFormData();
+    }
+    if (changes?.jjDisputedCount?.currentValue) {
+      this.jjDisputedCount = { ...this.jjDisputedCount, ...this.jjDisputedCount };
+      this.initFormData();
+    }
+    if (changes?.isSSEditMode?.currentValue) {
+      this.initFormData();
+    }
+  }
+
+  onSuspSentenceCheck(checked: boolean) {
+    if (!checked) {
+      this.form.get('jjDisputedCountRoP').get('ssProbationDuration').setValue(null);

```

Comment:
this.form.controls.jjDisputedCountRoP.ssProbationDuration

Reply:
this.form.controls.jjDisputedCountRoP.ssProbationDuration.setValue(null);
![image](https://github.com/bcgov/jag-traffic-courts-online/assets/55215368/946354d1-9e77-47f7-b8dd-c5bf090efee1)

Reply:
this.form.controls.jjDisputedCountRoP.controls.ssProbationDuration.setValue(null);

Reply:
If it is still not working just leave it and merge it

Reply:
Sorry, same error. controls doesn't exist on abstractControl.
Just an FYI, this code block is not part of this PR, this is original Lorraine code.

Reply:
ok because the diff doesn't work well, it is difficult to see